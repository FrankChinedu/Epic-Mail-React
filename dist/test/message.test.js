"use strict";var _supertest=_interopRequireDefault(require("supertest"));var _app=_interopRequireDefault(require("../app"));function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}const server=()=>(0,_supertest.default)(_app.default);let accessToken;const{apiURL}=global;describe('Message ',()=>{describe('/Post auth/login',()=>{it('should create an account for a new user',async()=>{const data={firstname:'aerrd',lastname:'angelo',email:'leap@me.com',password:'12345678'};const{status,body}=await server().post(`${apiURL}/auth/signup`).send(data);expect(Object.keys(body.data)).toMatchSnapshot();expect(status).toBe(201);});it('should log a user in',async()=>{const data={email:'leap@me.com',password:'12345678'};const{status,body}=await server().post(`${apiURL}/auth/login`).send(data);accessToken=body.data.token;expect(status).toBe(200);expect(Object.keys(body.data)).toMatchSnapshot();});});describe('/Post /messages',()=>{it('user should be able to create or send email',async()=>{const data={subject:'awesome subject',message:'thank God i am awesome',status:'draft',contactEmail:'paul@taru.com'};const{status,body}=await server().post(`${apiURL}/messages`).send(data).set('x-access-token',accessToken);expect(status).toBe(201);expect(Object.keys(body.data)).toMatchSnapshot();});});describe('/GET /messages',()=>{it('user should be able to get all received emails as inbox',async()=>{const{status,body}=await server().get(`${apiURL}/messages`).set('x-access-token',accessToken);expect(status).toBe(200);expect(Object.keys(body.data)).toMatchSnapshot();});});describe('/GET /messages/unread',()=>{it('user should be able to get all unread messages',async()=>{const{status,body}=await server().get(`${apiURL}/messages/unread`).set('x-access-token',accessToken);expect(status).toBe(200);expect(Object.keys(body.data)).toMatchSnapshot();});});describe('/GET /messages/:id',()=>{it('user should be able to get a specific user email from the inbox',async()=>{const{status,body}=await server().get(`${apiURL}/messages/:id`).set('x-access-token',accessToken);if(status===200){expect(status).toBe(200);expect(Object.keys(body.data)).toMatchSnapshot();}else{expect(status).toBe(404);expect(Object.keys(body.data)).toMatchSnapshot();}});});describe('/GET /messages/sent/:id',()=>{it('user should be able to get a specific user email from sent messages',async()=>{const{status,body}=await server().get(`${apiURL}/messages/sent/:id`).set('x-access-token',accessToken);if(status===200){expect(status).toBe(200);expect(Object.keys(body.data)).toMatchSnapshot();}else{expect(status).toBe(404);expect(Object.keys(body.data)).toMatchSnapshot();}});});describe('/DELETE /messages/:id',()=>{it('user should be able to delete a message from there inbox',async()=>{const{status,body}=await server().delete(`${apiURL}/messages/1`).set('x-access-token',accessToken);if(status===202){expect(status).toBe(202);expect(Object.keys(body.data)).toMatchSnapshot();}else{expect(status).toBe(404);expect(Object.keys(body.data)).toMatchSnapshot();}});});describe('/GET /messages/sent',()=>{it('user should be able to get all sent messages ',async()=>{const{status,body}=await server().get(`${apiURL}/messages/sent`).set('x-access-token',accessToken);expect(status).toBe(200);expect(Object.keys(body.data)).toMatchSnapshot();});});});