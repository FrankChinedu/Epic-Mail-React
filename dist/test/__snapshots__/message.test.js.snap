// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message  /DELETE /messages/:id user should be able to delete a message from there inbox 1`] = `
Array [
  "0",
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
]
`;

exports[`Message  /GET /messages user should be able to get all received emails as inbox 1`] = `Array []`;

exports[`Message  /GET /messages/:id user should be able to get a specific user email from the inbox 1`] = `Array []`;

exports[`Message  /GET /messages/sent user should be able to get all sent messages  1`] = `Array []`;

exports[`Message  /GET /messages/sent/:id user should be able to get a specific user email from sent messages 1`] = `Array []`;

exports[`Message  /GET /messages/unread user should be able to get all unread messages 1`] = `Array []`;

exports[`Message  /Post /messages user should be able to create or send email 1`] = `
Array [
  "id",
  "createdOn",
  "subject",
  "message",
  "parentMessageId",
  "status",
]
`;

exports[`Message  /Post auth/login should create an account for a new user 1`] = `
Array [
  "firstname",
  "lastname",
  "email",
  "token",
]
`;

exports[`Message  /Post auth/login should log a user in 1`] = `
Array [
  "firstname",
  "lastname",
  "email",
  "token",
]
`;
